File: app/src/main/java/com/volkanhotur/basemvvm/android/data/repository/Repository.java
package com.volkanhotur.basemvvm.android.data.repository;

import java.util.HashMap;

import io.reactivex.Observable;
import okhttp3.ResponseBody;
import retrofit2.Response;

public interface Repository {

    Observable<ResponseBody> postRequest(String url, HashMap<String, Object> params, HashMap<String, Object> headers);

    Observable<ResponseBody> postRequest(String url, HashMap<String, Object> params);

    Observable<ResponseBody> getRequest(String url, HashMap<String, Object> params, HashMap<String, Object> headers);

    Observable<ResponseBody> getRequest(String url);
}

File: app/src/main/java/com/volkanhotur/basemvvm/android/data/repository/Repository.kt
package com.volkanhotur.basemvvm.android.data.repository

import io.reactivex.Observable
import okhttp3.ResponseBody
import java.util.*

/**
 * @author volkanhotur
 */
interface Repository {

    fun postRequest(url: String?, params: HashMap<String?, Any?>?, headers: HashMap<String?, Any?>?): Observable<ResponseBody?>

    fun postRequest(url: String?, params: HashMap<String?, Any?>?): Observable<ResponseBody?>

    fun getRequest(url: String?, params: HashMap<String?, Any?>?, headers: HashMap<String?, Any?>?): Observable<ResponseBody?>

    fun getRequest(url: String?): Observable<ResponseBody?>
}